### PR TITLE
MGMT-14285: added validation to ApplianceConfig

### PR DESCRIPTION
**Pull request by danielerez**

### CODE REVIEW

File: pkg/asset/config/appliance_config.go [hunk 1]
```diff
@@ -138,63 +149,137 @@ func (a *ApplianceConfig) Load(f asset.FileFetcher) (bool, error) {
...
+				a.Config.OcpRelease.CpuArchitecture,
+				"specified OCP release cpuArchitecture is not supported ")}...)
+		}
+	}
+
+	return allErrs
+}
+
+func (a *ApplianceConfig) validateDiskSize() error {
+	if a.Config.DiskSizeGB < MinDiskSize {
```
Comment (nmagnezi) [inline pkg/asset/config/appliance_config.go:281]: note: we need to see what happens with float values

File: pkg/asset/config/appliance_config.go [hunk 2]
```diff
@@ -138,63 +149,137 @@ func (a *ApplianceConfig) Load(f asset.FileFetcher) (bool, error) {
...
+func (a *ApplianceConfig) validateApiVersion() field.ErrorList {
+	if a.Config.TypeMeta.APIVersion == "" {
+		return field.ErrorList{field.Required(field.NewPath("apiVersion"), "appliance-config version is required")}
+	}
+	switch v := a.Config.APIVersion; v {
+	case types.ApplianceConfigVersion: // Current version
+	default:
+		return field.ErrorList{field.Invalid(field.NewPath("apiVersion"),
+			a.Config.TypeMeta.APIVersion,
+			fmt.Sprintf("appliance-config version must be %q", types.ApplianceConfigVersion))}
```
[the remark this answers is not in the document]
Comment (danielerez) [reply pkg/asset/config/appliance_config.go]: done

File: pkg/asset/config/appliance_config.go (hunk 1 above)
Comment (danielerez) [reply pkg/asset/config/appliance_config.go:284]: Fails on json unmarshaling, maybe we can make it nicer later:
`FATAL failed to load asset "Appliance Config": failed to unmarshal appliance-config.yaml: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal number 200.2 into Go struct field ApplianceConfig.diskSizeGb of type int`

**Review by nmagnezi — APPROVED** [review]
/lgtm

**Pull request merged**